label_studio/tasks/models.py

```diff
@@ -992,6 +992,50 @@ class Meta:
         db_table = 'prediction'
 
 
+class FailedPrediction(models.Model):
+    """
+    Class for storing failed prediction(s) for a task
+    """
+    message = models.TextField(
+        _('message'),
+        default='',
+        blank=True,
+        null=True,
+        help_text='The message explaining why generating this prediction failed'
+    )
+    error_type = models.CharField(
+        _('error_type'),
+        max_length=512,
+        default=None,
+        null=True,
+        help_text='The type of error that caused prediction to fail'
+    )
+    ml_backend_model = models.ForeignKey(
+        'ml.MLBackend',
+        on_delete=models.SET_NULL,
+        related_name='failed_predictions',
+        null=True,
+        help_text='An ML Backend instance that created the failed prediction.',
+    )
+    model_version = models.TextField(
+        _('model version'),
+        default='',
+        blank=True,
+        null=True,
+        help_text='A string value that for model version that produced the failed prediction. Used in both live models and when uploading offline predictions.',
+    )
+    model_run = models.ForeignKey(
+        'ml_models.ModelRun',
+        on_delete=models.CASCADE,
+        related_name='failed_predictions',
+        null=True,
+        help_text='A run of a ModelVersion that created the failed prediction.',
+    )
+    project = models.ForeignKey('projects.Project', on_delete=models.CASCADE, related_name='failed_predictions', null=True)
```

Comment:
naive question: do we need the project if we have the task? Can't we get the project from the task?

Reply:
We could, but doesnt hurt to store, can get all failed predictions for a project without first getting a task and doing some weird queries. Tried to keep it similar to normal Prediction model so we can easily get both if/when needed